### ISSUE TITLE
Cannot connect using just public key auth

### ISSUE THREAD

**Issue by Ankcorn** [PowerShell/Win32-OpenSSH#1180]
Hi,
I cannot get it working with OpenSSH following the documentation listed [here](https://github.com/PowerShell/Win32-OpenSSH/wiki/ssh.exe-examples) 

Password auth works fine

**"OpenSSH for Windows" version**
 v7.7.0.0p1-Beta
**Server OperatingSystem**
Windows Server 2012
**Client OperatingSystem**
Windows Server 2016
**What is failing**
Cannot connect using public key auth. It always prompts me for my users password
**Expected output**
Connect with just public key auth
**Actual output**
Prompts for password

Apologies if there is a better forum for support. I have been stuck on this for a while

**Comment by manojampalam** [comment]
Without further details, it hard to guess what went wrong. Lets start of with traces, shttps://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Comment by Ankcorn** [comment]
Thanks

In the stdout it connects fine up until auth 

`debug1: Authentications that can continue: publickey,password,keyboard-interactive
debug1: Next authentication method: publickey
debug1: Offering public key: RSA SHA256:604SyK5C6dOapV6LGA7zli11Id8z8ISxa1jigX8EEUk ./id_rsa`

It attempts that with the path of the ./id_rsa file i gave it. Then the path to my users id_rsa key. Both which have been added to the local ssh agent and the public key is in the authorised key files with only Administrator and System permissions.

After attempting that it attempts password auth that works

On the sshd.exe stdout it shows it Could not open authorised keys <path to my authorised keys>

This is despite having opened up cmd as an administrator ran psexec -s cmd.

It sounds like I am doing something wrong with user permissions. Any advice would be appreciated 👍

**Comment by manojampalam** [comment]
Does it show an error code with 
Could not open authorised keys

**Comment by Ankcorn** [comment]
Exactly.

**Comment by manojampalam** [comment]
I meant does it show a numerical error code along with that message ?

**Comment by Ankcorn** [comment]
Oh sorry, unfortunately I cannot check till tomorrow morning. 

Thanks for your help :grinning:

**Comment by Ankcorn** [comment]
I have done a bit more detective work. 

The Error message is 
`
debug1: trying public key file C:\\Users\\<me>\\.ssh/authorized_keys
debug1: Could not open authorized keys 'C:\\Users\\<me>\\.ssh/authorized_keys': Permission Denied
Failed publickey for <me> from <localip> port 64842 ssh2: RSA SHA 256:<xxxxxxx>
`
I have also double checked my config to make sure PubkeyAuthentication is yes
The only similar issue I can find is https://github.com/PowerShell/Win32-OpenSSH/issues/287 but they feel unrelated. I still think I must have done something wrong but have followed the docs.


edit: Apologies for the accidentally closing and opening it.

**Comment by NoMoreFood** [comment]
@ThomasAnkcorn Can you attach the full debug log with DEBUG3?  If they aren't being produced, you can updated the lines in C:\ProgramData\ssh\sshd_config:

```
SyslogFacility LOCAL0
LogLevel DEBUG3
```

**Comment by Ankcorn** [comment]
@NoMoreFood Thanks for the help and apologies I didn't get back to you sooner. 

I have resolved my issue. It boiled down to me not understanding ssh well enough.
I copied my public key file into a folder called authorized_keys. Rather than the contents of the public key file into a file called authorized_keys.

To my defence I am not sure the documentation on this is that clear. It states:
> Copy id_rsa.pub (client's public key) to corresponding user's directory on the SSH server at %systemdrive%\Users\<user>\.ssh\authorized_keys

This could be ambiguous and hard to follow. Especially for other new users like me.

If it was changed to something like 
> Copy the content of id_rsa.pub (client's public key) to corresponding user's directory on the SSH server to a file with the path %systemdrive%\Users\<user>\.ssh\authorized_keys

It could be clearer.
Thanks again for all your help
